vernac/obligations.ml

```diff
@@ -20,6 +20,12 @@ open Pp
 open CErrors
 open Util
 
+(* EJGA: This should go away, no more need for fix_exn  *)
+module Stm = struct
+    let u = (fun x -> x)
+    let get_fix_exn () = u
+end
+
```

Comment:
the `Reach.known_state` function is recursive, if you have `s1 ---tac1--> s2 ---tac2--> s3` you know `s3` and you want to observe `s1` you recursively visit the DAG and run `tac2` then `tac1`.  So morally it is the exception that carries such piece of info.  If you make the effect algebraic, it would be something like ` OK | Fail(safe,err)` and Fail should propagate trough the call stack as an exception (or as `None` in the option monad).